docs/advanced-features/event-driven-packaging.md

```diff
@@ -0,0 +1,298 @@
+        "README.md": QUILT_README_STR,
+        ".quiltignore": QUILT_IGNORE_STR,
+    }
+
+    @logger.inject_lambda_context
+    def lambda_handler(event, context):
+
+        # EDP event data
+        bucket = event["detail"]["bucket"]
+        prefix = event["detail"]["prefix"]
+    
+        # Add every file in the prefix folder to the new data package
+        pkg = quilt3.Package().set_dir(".", f"s3://{bucket}/{prefix}")
+    
+        # Decorate the data package with example required metadata (as defined by WORKFLOW_NAME)
+        meta = {
+            "Author": "EDP",
+            "ComputerName": "Genome Lab - 1234",
+            "Date": datetime.date.today().strftime("%Y-%m-%d"),
+            "DiseaseArea": "COVID-19",
+            "Drug Substance": "RNA",
+            "ELN_ID": "827392-0102",
+            "FolderName": "A34-VB9877",
+            "Organism": "Monkey",
+            "ProjectID": "YYD",
+            "StudyID": "ABC-23-023394",
+            "Target": "T1",
+            "Test Material": "FIS-1111",
+            "Tissue": "Lungs"
+        }
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            tmpdir_path = pathlib.Path(tmpdir)
+            for name, body in beautify_files.items():
+                if name in pkg:
+                    logger.debug(f"File {name} already exists. Ignoring.")
+                    continue
+                logger.debug(f"File {name} does not exist at {prefix}. Creating.")
+                file_path = tmpdir_path / name
+                file_path.write_text(body)
+                pkg.set(name, file_path)
+        
+            # Add metadata to package
+            pkg.set_meta(meta)
+            # Remove leading & trailing characters
+            pkg_name = prefix.strip("/")
+        
+            # Define callable Quilt push()
+            push = functools.partial(
+                pkg.push,
+                pkg_name,
+                registry=f"s3://{bucket}",
+                force=True,
+                message="Created by EDP",
+                workflow=WORKFLOW_NAME
+            )
+        
+            # Validate against the Quilt workflow schema
+            try:
+                push(dedupe=True)
+            except quilt3.workflows.WorkflowValidationError as e:
+                logger.warning("Workflow check failed")
+            
+                # Write out error to README.md file in quarantine bucket
+                file_path = tmpdir_path / "README.md"
+                file_path.write_text(str(e))
+                pkg.set("README.md", file_path)
+            
+                # Push package to quarantine bucket
+                push(registry=f"s3://{QUARANTINE_BUCKET_NAME}", workflow=...)
+            
+                # Error SNS notification content
+                subject = f"Failed validation for package {pkg_name}"
+                if len(subject) > MAX_SNS_SUBJECT_LEN:
+                    subject = subject[: MAX_SNS_SUBJECT_LEN - 1] + "…"
```

Comment:
It would be simpler to limit the subject to "Failed to create package" and rely on the message for everything else (eliminates at least on variable and several lines of code).